--- a/src/python/setup.py.in
+++ b/src/python/setup.py.in
@@ -71,14 +71,14 @@
   INCLUDES.append(os.path.join(dbxml_home,"include"))
   INCLUDES.append(os.path.join(db_home, "include"))
 
-  LIBDIRS =     [os.path.join(dbxml_home,"build_unix/.libs"),
+  LIBDIRS =     [os.path.join(dbxml_home,".libs"),
                  os.path.join(dbxml_home, "lib"),
                  os.path.join(db_home, "lib"),
                  os.path.join(xqilla_home, "lib"),
                  os.path.join(xerces_home, "lib")]
 
   LIBS =        ["dbxml",
-                 "db-4",
+                 "db-5.1",
                  "xqilla", "xerces-c"]
 
   DATAFILES =   []
--- a/dist/swig/dbxml_python.i
+++ b/dist/swig/dbxml_python.i
@@ -491,11 +491,11 @@
 }
 
 %init %{
-	static const int dbMajorMin=4;
-	static const int dbMinorMin=3;
+	static const int dbMajorMin=5;
+	static const int dbMinorMin=1;
 	int major, minor, patch, dmajor, dminor, dpatch;
 	(void) db_version(&major, &minor, &patch);
-	if (major < dbMajorMin || minor < dbMinorMin) {
+	if (major < dbMajorMin) {
 		char buf[512];
 		PyOS_snprintf(buf, sizeof(buf), "Version mismatch: Berkeley DB XML %d.%d.%d requires Berkeley DB \n%d.%d or higher; version %d.%d.%d was found\n",
 			      DBXML_VERSION_MAJOR, DBXML_VERSION_MINOR,
--- a/src/python/dbxml_python_wrap.cpp
+++ b/src/python/dbxml_python_wrap.cpp
@@ -42261,11 +42261,11 @@
   
   
   
-  static const int dbMajorMin=4;
-  static const int dbMinorMin=3;
+  static const int dbMajorMin=5;
+  static const int dbMinorMin=1;
   int major, minor, patch, dmajor, dminor, dpatch;
   (void) db_version(&major, &minor, &patch);
-  if (major < dbMajorMin || minor < dbMinorMin) {
+  if (major < dbMajorMin) {
     char buf[512];
     PyOS_snprintf(buf, sizeof(buf), "Version mismatch: Berkeley DB XML %d.%d.%d requires Berkeley DB \n%d.%d or higher; version %d.%d.%d was found\n",
       DBXML_VERSION_MAJOR, DBXML_VERSION_MINOR,
